### ISSUE TITLE
epoll_ctl failure

### ISSUE THREAD

**Issue by collinhundley** [IBM-Swift/Kitura#708]
Every 10-20 requests, the following error gets logged:

```sh
ERROR: removeIdleSockets() IncomingSocketManager.swift line 166 - epoll_ctl failure. Error code=9. Reason=Bad file descriptor
```

I don't know what the error means, and I haven't found any apparent cause for it. The request still seems to succeed even when the error does get logged. Any ideas?

**Comment by dsperling** [comment]
I am seeing this issue now as well.  It is fairly new, but I am have not identified which version it was introduced.

**Comment by shmuelk** [comment]
The code at that point is trying to close idle sockets that were kept open due to keep alive. My guess is that somehow the socket in question got closed somehow or other, and Kitura-net somehow missed that fact. Perhaps in the case of an epoll_ctl error with errno = EBADF (9), we should fail silently.

**Comment by collinhundley** [comment]
Did this end up making its way into 0.32?

**Comment by shmuelk** [comment]
No it didn't. We will have another release next week. It will be in that release.

**Comment by shmuelk** [comment]
@collinhundley The fix is in Kitura 0.33
